### PR TITLE
Cool stuff

### PR DESCRIPTION
**Pull request by big-lip-bob**

### CODE REVIEW

**Comment by big-lip-bob** [comment]
If the FreeAll command is limited by the discord API rates, wouldn't using some of the JDA utulities to wait the threshold out be possible

File: src/main/java/org/learn_java/bot/data/entities/Warn.java [hunk 1]
```diff
@@ -1,44 +1,32 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
 
 @Entity
 public class Warn {
   @Id @GeneratedValue private long id;
-  private String username;
-  private String reason;
-  private LocalDate date;
+  private long user_id;
```
Comment (Kemikals) [inline src/main/java/org/learn_java/bot/data/entities/Warn.java]: this and every variable like it needs to be renamed to java standards

File: src/main/java/org/learn_java/bot/data/entities/Warn.java [hunk 2]
```diff
@@ -1,44 +1,32 @@
...
 
-  public Warn(String username, String reason, LocalDate date) {
-    this.username = username;
+  public Warn(long username, String reason, LocalDateTime date) {
+    this.id = username;
     this.reason = reason;
     this.date = date;
   }
 
-  public Warn() {}
```
Comment (Kemikals) [inline src/main/java/org/learn_java/bot/data/entities/Warn.java:19]: This is needed with spring

File: src/main/java/org/learn_java/bot/data/entities/Warn.java [hunk 3]
```diff
@@ -1,44 +1,32 @@
...
-
-  public String getUsername() {
```
Comment (Kemikals) [inline src/main/java/org/learn_java/bot/data/entities/Warn.java:21]: Along with all of these getters/setters

File: src/main/java/org/learn_java/bot/data/repositories/BanRepository.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.learn_java.bot.data.repositories;
+
+import java.util.List;
+import org.learn_java.bot.data.entities.Ban;
+import org.springframework.data.jpa.repository.JpaRepository;
+// import org.springframework.stereotype.Repository;
+
+// @Repository
+public interface BanRepository extends JpaRepository<Ban, Long> {
+  List<Ban> findByUser_ID(long user_id);
```
Comment (Kemikals) [inline src/main/java/org/learn_java/bot/data/repositories/BanRepository.java]: Rename this once you fix the field name as well

File: src/main/java/org/learn_java/bot/commands/free/NewFree.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import com.jagrosh.jdautilities.command.Command;
+import com.jagrosh.jdautilities.command.CommandEvent;
+import com.vdurmont.emoji.EmojiManager;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.managers.ChannelManager;
+import org.springframework.stereotype.Component;
+
+@Component
+public class NewFree extends Command {
```
Comment (Kemikals) [inline src/main/java/org/learn_java/bot/commands/free/NewFree.java]: What is the point of this class?

File: src/main/java/org/learn_java/bot/data/entities/Ban.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.time.LocalDateTime;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+@Entity
+public class Ban {
+  @Id @GeneratedValue private long id;
+  private final long user_id;
```
Comment (Kemikals) [inline src/main/java/org/learn_java/bot/data/entities/Ban.java]: Java naming conventions

File: src/main/java/org/learn_java/bot/data/entities/Ban.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    this.is_premanent = false;
+    this.unban_date = unban_date;
+  }
+
+  public Ban(long id, String reason) {
+    this.user_id = id;
+    this.reason = reason;
+    this.is_premanent = true;
+  }
+
```
Comment (Kemikals) [inline src/main/java/org/learn_java/bot/data/entities/Ban.java:30]: Also need a no arg constructor, and getters/setters for all fields

**Review by Kemikals — COMMENTED** [review]
One big thing is why are you trying to implement 4 new features and refactors that touch multiple things in one big PR. The Ban repository isn't even used in this. Split this up with 1 feature per PR, and name it something meaningful not Cool stuff,

**Comment by Kemikals** [comment]
> If the FreeAll command is limited by the discord API rates, wouldn't using some of the JDA utulities to wait the threshold out be possible

In short there is no way to tell until you get the reply and even then you don't know which it's for. Basically it's per channel and there's no way to tell because of how the Discord API is. You can assume, but you can't know for sure.

File: src/main/java/org/learn_java/bot/data/entities/Ban.java (hunk 6 above)
Comment (big-lip-bob) [reply src/main/java/org/learn_java/bot/data/entities/Ban.java]: What is java

File: src/main/java/org/learn_java/bot/data/entities/Ban.java (hunk 7 above)
Comment (big-lip-bob) [reply src/main/java/org/learn_java/bot/data/entities/Ban.java:30]: Got it

File: src/main/java/org/learn_java/bot/commands/free/NewFree.java (hunk 5 above)
Comment (big-lip-bob) [reply src/main/java/org/learn_java/bot/commands/free/NewFree.java]: I guess ill remove the freeing stuff cause of discord's API, and i cant' be bothered to implement a queue

File: src/main/java/org/learn_java/bot/data/repositories/BanRepository.java (hunk 4 above)
Comment (big-lip-bob) [reply src/main/java/org/learn_java/bot/data/repositories/BanRepository.java]: Rename to ? haven't had the opportunity to work with databases so

File: src/main/java/org/learn_java/bot/data/entities/Warn.java (hunk 2 above)
Comment (big-lip-bob) [reply src/main/java/org/learn_java/bot/data/entities/Warn.java:19]: Got it

File: src/main/java/org/learn_java/bot/data/entities/Warn.java (hunk 3 above)
Comment (big-lip-bob) [reply src/main/java/org/learn_java/bot/data/entities/Warn.java:21]: Got it

**Pull request closed without merging**